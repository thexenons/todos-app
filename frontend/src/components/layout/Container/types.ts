import { PropsWithChildren } from "react";

export interface ContainerProps extends PropsWithChildren {
	fullHeight?: boolean;
}
