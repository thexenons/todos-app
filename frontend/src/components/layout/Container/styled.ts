import styled from "styled-components";

export const ContainerWrapper = styled.div`
  width: 100%;

  max-width: ${({ theme }) => theme.components.layout.container.maxWidth}px;

  ${({ theme }) => theme.mixins.padding({ y: 0, x: "var(--page-padding)" })}
  ${({ theme }) => theme.mixins.margin({ y: 0, x: "auto" })}
`;
