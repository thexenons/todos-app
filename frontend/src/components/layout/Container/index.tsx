import type { FC, PropsWithChildren } from "react";
import * as S from "./styled";

const Container: FC<PropsWithChildren> = ({ children }) => (
  <S.ContainerWrapper>{children}</S.ContainerWrapper>
);

export default Container;
