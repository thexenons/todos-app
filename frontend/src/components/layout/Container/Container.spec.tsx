import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import Container from ".";

describe("Container", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<Container />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
