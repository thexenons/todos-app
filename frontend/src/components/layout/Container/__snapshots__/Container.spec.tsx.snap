// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > Render 1`] = `
<div>
  <div
    class="styled__ContainerWrapper-sc-6862f76b-0 jasIgO"
  />
</div>
`;
